modules/m_whois.c

```diff
@@ -269,29 +266,38 @@ single_whois(struct Client *source_p, struct Client *target_p, int operspy)
 
 	if (!IsService(target_p))
 	{
-		RB_DLINK_FOREACH(ptr, target_p->user->channel.head)
+		hook_data_channel_visibility hdata_vis;
+		rb_dlink_node *ps, *pt;
+		struct Channel *chptr;
+		struct membership *ms, *mt;
+
+		hdata_vis.client = source_p;
+
+		ITER_COMM_CHANNELS(ps, pt, source_p->user->channel.head, target_p->user->channel.head, ms, mt, chptr)
 		{
-			msptr = ptr->data;
-			chptr = msptr->chptr;
+			if (mt == NULL)
+				continue;
 
-			hdata.chptr = chptr;
+			hdata_vis.chptr = chptr;
+			hdata_vis.clientms = ms;
+			hdata_vis.targms = mt;
```

Comment:
To go with the struct change:

```suggestion
			hdata_vis.targetms = mt;
```